linux/src/main/java/com/github/weisj/darkmode/platform/linux/gnome/GnomeGtkThemeMonitor.java

```diff
@@ -0,0 +1,172 @@
+package com.github.weisj.darkmode.platform.linux.gnome;
+
+import com.github.weisj.darkmode.platform.ThemeMonitorService;
+import com.github.weisj.darkmode.platform.linux.util.ProcessResult;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.github.weisj.darkmode.platform.LibraryUtil.userHomeDirectory;
+import static com.github.weisj.darkmode.platform.linux.util.CmdlineUtils.exec;
+import static java.util.Objects.isNull;
+
+/**
+ * This class monitors the GTK theme which is set for the current Gnome environment and makes an estimated guess as to
+ * whether the GTK theme is meant to resemble a "light theme", "dark theme", and whether the theme is "high contrast".
+ * <p>
+ * TODO: Remove the guessing feature in favor of a settings panel where users can specify their themes themselves.
+ */
+public class GnomeGtkThemeMonitor implements Runnable {
+    private final static Logger logger = Logger.getLogger(GnomeGtkThemeMonitor.class.getName());
```

Comment:
Logger should be static. It might also be worth it to add a custom `Logger` interface (Doesn't need to be fully fletched. error/info/warning methods should be enough) to the `base` module with an own static factory method that delegated to `Logger.getInstance()` from the IntelliJ Platform api. This way the log messages are saved to the intellij log and errors are made visible to the user.